orchestrator/careplancontributor/ehr/kafka_client.go

```diff
@@ -0,0 +1,169 @@
+//   - Topic: The Kafka topic to which messages will be sent.
+//   - Endpoint: The Kafka broker endpoint.
+//   - ConnectionString: The connection string used for authentication.
+type KafkaConfig struct {
+	Enabled  bool           `koanf:"enabled" default:"false"`
+	Topic    string         `koanf:"topic"`
+	Endpoint string         `koanf:"endpoint"`
+	Sasl     SaslConfig     `koanf:"sasl"`
+	Security SecurityConfig `koanf:"security"`
+}
+
+type SaslConfig struct {
+	Mechanism string `koanf:"mechanism" default:"PLAIN"`
+	Username  string `koanf:"username" default:"$ConnectionString"`
+	Password  string `koanf:"password"`
+}
+
+type SecurityConfig struct {
+	Protocol string `koanf:"protocol" default:"SASL_PLAINTEXT"`
+}
+
+// KafkaClient is an interface that defines the method for submitting messages.
+// Implementations of this interface should provide the logic for submitting
+// messages to a Kafka topic or an alternative storage.
+//
+// Methods:
+//   - SubmitMessage: Submits a message with the given key and value.
+//
+// Parameters:
+//   - key: The key of the message to be submitted.
+//   - value: The value of the message to be submitted.
+//
+// Returns:
+//   - error: An error if the message could not be submitted.
+type KafkaClient interface {
+	SubmitMessage(key string, value string) error
+}
+
+// KafkaClientImpl is an implementation of the KafkaClient interface.
+// It holds the Kafka topic and producer used to submit messages.
+type KafkaClientImpl struct {
+	topic    string
+	producer *kafka.Producer
+}
+
+type NoopClientImpl struct {
+}
+
+// NewClient creates a new KafkaClient based on the provided KafkaConfig.
+// If Kafka is enabled in the configuration, it initializes a Kafka producer
+// and sets up a delivery report handler for produced messages. If Kafka is
+// not enabled, it returns a NoopClientImpl which writes messages to a file.
+//
+// Parameters:
+//   - config: KafkaConfig containing the configuration for the Kafka client.
+//
+// Returns:
+//   - KafkaClient: An implementation of the KafkaClient interface.
+//   - error: An error if the Kafka producer could not be created.
+func NewClient(config KafkaConfig) (KafkaClient, error) {
+
+	var kafkaClient KafkaClient
+	if config.Enabled {
+		endpoint := config.Endpoint
+		log.Infof("KafkaClientImpl, connecting to %s, Mechanism: %s, protocol: %s, username: %s, password: %s",
+			endpoint,
+			config.Sasl.Mechanism,
+			config.Security.Protocol,
+			config.Sasl.Username,
+			config.Sasl.Password)
+		producer, err := kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers": endpoint,
+			"sasl.mechanisms":   config.Sasl.Mechanism,
+			"sasl.username":     config.Sasl.Username,
+			"sasl.password":     config.Sasl.Password,
+			"security.protocol": config.Security.Protocol,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Delivery report handler for produced messages
+		go func() {
+			log.Infof("Kafka func started")
```

Comment:
debug logging, probably can be removed?

Reply:
I'd agree, remove this log, make the one below Debug. You'll know your Kafka func has started when you start receiving events